src/opentime/rationalTime.cpp

```diff
@@ -13,54 +13,33 @@ namespace opentime { namespace OPENTIME_VERSION {
 
 RationalTime RationalTime::_invalid_time{ 0, RationalTime::_invalid_rate };
 
-static constexpr std::array<double, 4> dropframe_timecode_rates{ {
-    // 23.976,
-    // 23.98,
-    // 23.97,
-    // 24000.0/1001.0,
-    29.97,
+static constexpr std::array<double, 2> dropframe_timecode_rates{ {
     30000.0 / 1001.0,
-    59.94,
     60000.0 / 1001.0,
 } };
 
+// See the official source of these numbers here:
+// ST 12-1:2014 - SMPTE Standard - Time and Control Code
+// https://ieeexplore.ieee.org/document/7291029
+//
 static constexpr std::array<double, 11> smpte_timecode_rates{
-    { 1.0,
-      12.0,
-      24000.0 / 1001.0,
+    { 24000.0 / 1001.0,
       24.0,
       25.0,
       30000.0 / 1001.0,
       30.0,
+      48000.0 / 1001.0,
       48.0,
       50.0,
       60000.0 / 1001.0,
-      60.0 }
-};
-
-static constexpr std::array<double, 16> valid_timecode_rates{
-    { 1.0,
-      12.0,
-      23.97,
-      23.976,
-      23.98,
-      24000.0 / 1001.0,
-      24.0,
-      25.0,
-      29.97,
-      30000.0 / 1001.0,
-      30.0,
-      48.0,
-      50.0,
-      59.94,
-      60000.0 / 1001.0,
-      60.0 }
+      60.0
+    }
 };
 
 bool
 RationalTime::is_valid_timecode_rate(double fps)
 {
-    auto b = valid_timecode_rates.begin(), e = valid_timecode_rates.end();
+    auto b = smpte_timecode_rates.begin(), e = smpte_timecode_rates.end();
```

Comment:
Hopefully I am not opening a can of worms here... 

I see we were already using std::find (which uses operator==) before your changes, but might we want to put comparison with epsilon instead for floating point tolerance?

```
std::find_if(b, e,[fps](double v) { return std::fabs(fps - v) < std::numeric_limits<double>::epsilon(); });
```

The `nearest_valid_timecode_rate` function could be used as well for readability, although it's likely not as efficient:

```
return std::fabs(fps - nearest_valid_timecode_rate(fps)) < std::numeric_limits<double>::epsilon();
```

Reply:
Agreed that exact comparisons are not as robust as tolerance based comparisons. `std::numeric_limits<double>::epsilon()` per se probably doesn't help because it is to small a value to catch inaccuracies from typical calculations (you lose a bit here and there after every math op, so imprecision is typically likely going to exceed epsilon by a factor on the order of the number of intervening calculations), although feel free to challenge my reasoning :) 

The highest framerate we test for is `60000.0 / 1001.0` so the choice I'd recommend for a tolerance is half that, e.g. `120000.0 / 1001.0` since that's double the max rate we check for.

Reply:
I agree with you it makes sense to use a higher tolerance here, especially since we're supporting bindings to many different languages, and I've seen floating pointing numbers go awry just from crossing through language barriers.

So something like this should do it:
```
// Use half the highest frame rate as our max tolerance
static constexpr MAX_FPS_TOLERANCE = 120000.0 / 1001.0;
std::find_if(b, e,[fps](double v) { return std::fabs(fps - v) < MAX_FPS_TOLERANCE; });
```

Reply:
I like that :)

Reply:
I just noticed during the TSC meeting, I think we mean max tolerance = 1001.0 / 120000.0 right?

Reply:
I'm going to take the position here that any snapping or tolerance of slightly-off values should be handled via `nearest_smpte_timecode_rate` so that we can avoid propagating error into further calculations. Yes, those slightly-off cases will come up, but we want folks to use these functions to get the best value OTIO can provide, not to allow a slightly-off value to be used.

Reply:
That makes snese.  Perhaps then this method should return true/false based on whether or not `nearest_smpte_timecode_rate` is within the standard c++ tolerance for double comparison then so we don't run into the classic floating point comparison issues?

Simply:
```
return std::fabs(fps - nearest_smpte_timecode_rate(fps)) < std::numeric_limits<double>::epsilon();
```

The below is an additional thought, but I would not say it's required, I leave it up to you.

Note that there is already a hardcoded built in tolerance for  `nearest_smpte_timecode_rate` of 0.1 happening on line 461.

Perhaps to also address Nick's tolerance concerns for frame rates we could have a method like this that could be called there and used to replace that block of code:
```
double snap_to_smpte_rate(double rate, ErrorStatus* status)
{
   static constexpr double MAX_RATE_TOLERANCE = 1001.0 / 120000.0;
   const double nearest_rate = nearest_smpte_timecode_rate(rate);

   if (abs(nearest_rate) - rate) < MAX_RATE_TOLERANCE) return nearest_rate;
 
   if (error_status)
   {
      *error_status = ErrorStatus(ErrorStatus::INVALID_TIMECODE_RATE);
   }
   return 0.0;
}
```